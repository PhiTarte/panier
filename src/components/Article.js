import React from 'react'
import { Button, Col, ListGroupItem, Row } from 'reactstrap'

const Article = ({imgSrc, name, price}) => <ListGroupItem>
    <Row>
        <Col xs={3}>
            <Button>Supp</Button>
        </Col>
        <Col xs={3}>
            <img className="img-fluid" alt={ name } src={ imgSrc } />
        </Col>
        <Col xs={3}>
            <h3>{ name }</h3>
        </Col>
        <Col xs={3}>
            <p>{ price } â‚¬</p>
        </Col>
    </Row>
</ListGroupItem>

export default Article
