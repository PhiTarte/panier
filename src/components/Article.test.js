import React from 'react';
import ReactDOM from 'react-dom';
import Article from './Article';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Article />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders Article html', () => {
  
  const props = {
    name: "Benzema",
    price: 150000000,
    imgSrc: "https://i0.wp.com/observalgerie.com/wp-content/uploads/2018/05/2170922-45375850-2560-1440.jpg?fit=1050%2C590&ssl=1"
  }
  
  const tree = renderer
    .create(<Article {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
