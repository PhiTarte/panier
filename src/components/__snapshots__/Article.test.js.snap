// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Article html 1`] = `
<li
  className="list-group-item"
>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        type={undefined}
      >
        Supp
      </button>
    </div>
    <div
      className="col-3"
    >
      <img
        alt="Benzema"
        className="img-fluid"
        src="https://i0.wp.com/observalgerie.com/wp-content/uploads/2018/05/2170922-45375850-2560-1440.jpg?fit=1050%2C590&ssl=1"
      />
    </div>
    <div
      className="col-3"
    >
      <h3>
        Benzema
      </h3>
    </div>
    <div
      className="col-3"
    >
      <p>
        150000000
         â‚¬
      </p>
    </div>
  </div>
</li>
`;
